src/__experimental__/components/decimal/decimal.component.js

```diff
@@ -0,0 +1,123 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Textbox from '../textbox';
+import I18nHelper from '../../../utils/helpers/i18n';
+import Logger from '../../../utils/logger';
+
+class Decimal extends React.Component {
+  input = null;
+
+  // Create ref to document for tests
+  _document = document;
+
+  formatValue = () => {
+    const { value } = this.props;
+    const { input } = this;
+
+    // Return unformatted value if component has not mounted
+    if (!input) {
+      return value;
+    }
+    // Return unformatted value if input is being used
+    if (input && this._document.activeElement === input.current) {
+      return value;
+    }
+
+    // Only format value if input is not active
+    // Strip delimiters otherwise formatDecimal Helper goes nuts
+    const format = I18nHelper.format();
+    const delimiter = `\\${format.delimiter}`;
+    const removeDelimiter = new RegExp(`[${delimiter}]*`, 'g');
```

Comment:
that name suggests it's a method

Reply:
Again, I'm happy with this name, as it gives an indication as to how the regex is going to be used.

Reply:
I'm inclined to agree with Kuba, something like `delimiterMatcher` is much more intuitive. 

There's nothing inherent in that regular expression that connects it to deletion, it's just a matcher.